tests/baselines/reference/generatorReturnTypeInference.types

```diff
@@ -0,0 +1,308 @@
+>iterableIterator : IterableIterator<number>
+
+declare const generator: Generator<number, string, boolean>;
+>generator : Generator<number, string, boolean>
+
+declare const never: never;
+>never : never
+
+function* g000() { // Generator<never, void, unknown>
+>g000 : () => Generator<never, void, unknown>
+}
+
+// 'yield' iteration type inference
+function* g001() { // Generator<undefined, void, unknown>
+>g001 : () => Generator<undefined, void, unknown>
+
+    yield;
+>yield : any
+}
+
+function* g002() { // Generator<number, void, unknown>
+>g002 : () => Generator<number, void, unknown>
+
+    yield 1;
+>yield 1 : any
+>1 : 1
+}
+
+function* g003() { // Generator<never, void, unknown>
+>g003 : () => Generator<never, void, undefined>
+
+    yield* [];
+>yield* [] : any
+>[] : never[]
+}
+
+function* g004() { // Generator<number, void, undefined>
+>g004 : () => Generator<number, void, undefined>
+
+    yield* iterableIterator;
+>yield* iterableIterator : any
+>iterableIterator : IterableIterator<number>
+}
+
+function* g005() { // Generator<number, void, boolean>
+>g005 : () => Generator<number, void, boolean>
+
+    const x = yield* generator;
+>x : string
+>yield* generator : string
+>generator : Generator<number, string, boolean>
+}
+
+function* g006() { // Generator<1 | 2, void, unknown>
+>g006 : () => Generator<1 | 2, void, unknown>
+
+    yield 1;
+>yield 1 : any
+>1 : 1
+
+    yield 2;
+>yield 2 : any
+>2 : 2
+}
+
+function* g007() { // Generator<never, void, unknown>
+>g007 : () => Generator<never, void, unknown>
+
+    yield never;
+>yield never : any
+>never : never
+}
+
+// 'return' iteration type inference
+function* g102() { // Generator<never, number, unknown>
+>g102 : () => Generator<never, number, unknown>
+
+    return 1;
+>1 : 1
+}
+
+function* g103() { // Generator<never, 1 | 2, unknown>
+>g103 : () => Generator<never, 1 | 2, unknown>
+
+    if (Math.random()) return 1;
+>Math.random() : number
+>Math.random : () => number
+>Math : Math
+>random : () => number
+>1 : 1
+
+    return 2;
+>2 : 2
+}
+
+function* g104() { // Generator<never, never, unknown>
+>g104 : () => Generator<never, never, unknown>
+
+    return never;
+>never : never
+}
+
+// 'next' iteration type inference
+function* g201() { // Generator<number, void, string>
+>g201 : () => Generator<number, void, string>
+
+    let a: string = yield 1;
+>a : string
+>yield 1 : any
+>1 : 1
+}
+
+function* g202() { // Generator<1 | 2, void, never>
+>g202 : () => Generator<1 | 2, void, never>
+
+    let a: string = yield 1;
+>a : string
+>yield 1 : any
+>1 : 1
+
+    let b: number = yield 2;
+>b : number
+>yield 2 : any
+>2 : 2
+}
+
+declare function f1(x: string): void;
+>f1 : { (x: string): void; (x: number): void; }
+>x : string
+
+declare function f1(x: number): void;
+>f1 : { (x: string): void; (x: number): void; }
+>x : number
+
+function* g203() { // Generator<number, void, string>
+>g203 : () => Generator<number, void, string>
+
+	const x = f1(yield 1);

```

Comment:
why infer string from the contextual type? we usually take the last overload when we can't decide between signatures, so I think something more complicated is happening.

Reply:
I thought we take the first overload? I'm not doing anything unique here, just calling into our existing contextual typing behavior, so it should be consistent.